[CS2113-T12-4] WellNUS++

wenxin-c opened the pull request:
WellNUS++ is a Command Line Interface(CLI) app for **NUS Computing students** to keep track and improve their physical and mental wellness in various aspects. If you can type fast, WellNUS++ can update their wellness progress faster than traditional Graphical User Interface(GUI) apps.

ngkaiwen123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+
+### Reflection Component
+
+![Reflection Component Class Diagram](diagrams/ReflectionSequenceDiagram.png)
+![Reflection Component Class Diagram](diagrams/ReflectionClassDiagram.png)
```
![image](https://user-images.githubusercontent.com/88386677/228424892-f36f4d81-ca86-4957-8c7f-504712b78015.png)
The dotted arrows appear to be a little confusing. Perhaps it would be possible to redraw them?

Stella1585 replied:
The association lines are all overlapping, and it is difficult to tell them apart, especially at the top where the label and lines overlap. once again i suggest you can separate into perhaps feature and command diagrams.

ngyida replied:
This diagram is too crowded which makes it hard to read. Could you perhaps abstract some of the details out?

shawntangy replied:
The class diagram here could possibly be abstracted out and separated out into different features since it is not mandatory to include everything. Users may find it hard to read as the text size is a little small

javienneyeo replied:
diagram is messy and hard to visualise. might want to consider removing some of the less important parts and only showing the important features of the diagram

ngkaiwen123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+- A `QuestionList` object is passed in as a dependency to generate the set of indexes and provide the pool of 10
+  introspective questions available.
+
+`LikeCommand` class: <br>
+
+- Command format: `like <index of question(1~5)>`
+- Users can add an introspective question that is generated in the previous set into their favorite list. Since there
+  will only
+  be 5 questions per set, the indexes are restricted to integer 1~5.
+- `addFavQuestion()` method in `QuestionList` class is used to add and store the data.
+- Users can only successfully add a question to favorite list if they have gotten a set of questions previously.
+- Every time a question is added into the favorite list, the indexes of this particular question will be stored in data
+  file straightaway. It prevents data loss due to unforeseen computer shutdown.
+
+`FavoriteCommand` class: <br>
+
+- Command format: `fav`
+- Users can get the questions in their favorite list.
+- `getFavQuestions()` method in `QuestionList` class matches the indexes to corresponding questions and return this set
+  of questions
+  back to `FavCommand` for output. As such, `QuestionList` is a dependency for `FavoriteCommand` as well.
+
+`HomeCommand` class: <br>
+
+- Command format: `home`
+- This command allows users to return back to the main WellNUS++ interface.
+- Similar to `GetCommand`, `validateCommand()` method will also be called to validate the command.
+- It will then call the class-level method `ReflectionManager.setIsExit()` to terminate the while loop
+  in `Reflectionmanager`.
```
Perhaps it will also be great to include how these classes handle errors? Or if there are any cases (or zero cases) of errors they may need to handle?

ngkaiwen123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Implementation of `parseUserInput`:
 
+![CommandParser implementation](diagrams/CommandParserSequence.png)
```
![image](https://user-images.githubusercontent.com/88386677/228426028-93acd900-f0f3-4e02-b46a-f1484675f0a8.png)
A reference frame for the CommandParser accessing its own methods inside the loop may help to better describe what is happening in the sequence diagram?

Stella1585 replied:
Well made sequence diagram, helps me to understand how a command is parsed.

ngkaiwen123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+
+### AtomicHabit Component
+
+![AtomicHabit Component](diagrams/AtomicHabit.png)
```
![image](https://user-images.githubusercontent.com/88386677/228427007-a1b27ccb-120d-4612-845b-63a06a0f3df4.png)
A redraw of the arrows to make them seem neater may be better for the reader?

shawntangy replied:
To simplify the diagram, maybe not every command has to be shown, which would definitely reduce the amount of arrows

ngkaiwen123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+#### Overview
+The overall execution lifecycle of the WellNus application involves 4 main components, as shown in the diagram below.
+
+![Application Lifecycle](diagrams/WellnusSequence.png)
```
![image](https://user-images.githubusercontent.com/88386677/228428450-1445a4de-74da-4c52-b35c-76ceed446ea0.png)
Text on dotted lines should be the variable returned rather than a description?

ngkaiwen123 reviewed (commented):
Description of design and implementation is generally well done! Good work! Some diagrams may need to be improve for better readability.

Stella1585 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+
+### Reflection Component
+
+![Reflection Component Class Diagram](diagrams/ReflectionSequenceDiagram.png)
```
I feel that this might be a little repetitive as each command is executed in a similar way. It also ends up being very long and looks complicated. Maybe you can separate the commands.

Stella1585 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+The application begins with a call to `WellNus.start()`, which initialises an instance of `MainManager` and calls the 
+`MainManager.runEventDriver()` method.
+
+`MainManager.runEventDriver()` will then take control of user input and provide a basic interface that parses commands 
```
There are mentions of keywords and user inputs that are being tackled but i do not see a string being passed through or a text ui class, maybe u can show this in your digram too?

ngyida commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+
+![Application Lifecycle](diagrams/WellnusSequence.png)
+
+The application begins with a call to `WellNus.start()`, which initialises an instance of `MainManager` and calls the 
```
Could this be done without since it is a description of the sequence diagram?

ngyida commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+are not the focus of this section since they are outside of `reflection` package.<br>
+<br>
+
+#### Feature Package (`ReflectionManager`, `ReflectionQuestion`, `QuestionList`, `TextUi`, `RandomNumberGenerator` classes)
```
Could consider the use of UML diagrams to represent class to improve clarity

ngyida commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+
+#### Integration with WellNUS++
+
+![Integration](diagrams/CommandParserClass.png)
```
This class diagram may not follow the standard UML notation

ngyida reviewed (commented):
Nice DG overall! The link to github is a nice touch :)

shawntangy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+  The payload will terminate when the user clicks `enter` or separates the payload with another argument
+  with the `--` delimiter.
 
 ## Instructions for manual testing
```
Maybe this section could be removed from the developer guide? This is because your user guide already covers the commands, creating duplicate content.

shawntangy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+
+For example,
+`
+$ foo bar --arg1 payload1 payload1--1 --arg2 payload2 --arg3
```
Maybe this could be included in a code block for easier visualization

shawntangy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+* The app is gamified to make it more attractive for students to use. Levels and micro-goals incentivise our
+  users to keep using the app’s features, allowing them to focus on their work and achieve wellness.
 
 ## User Stories
```
User stories language format could be changed to fit the "I want to ..." sentence. For e.g. I want to use keyboard instead of mouse vs I want to I can use keyboard instead of mouse

javienneyeo commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,726 @@
+| v2.0    | Reflective student                                 | I can get the previous questions I viewed                   | I can re-view these questions                      |
+| v2.0    | Easily distracted computing student                | I want to start a timer to keep track of time spent on work | I can do timed-practice                            |
+| v2.0    | Easily distracted computing student                | I want to check the time                                    | I can keep track of my pace                        |
+| v2.0    | A regular WellNUS++ user                           | I wish to have my information stored in the app             | I can re-view my past data                         |
```
can add priority levels / level of importance to this

Stella1585 reviewed (commented):
(sorry i forgot to submit my review)
developer guide is very informative!

YongbinWang committed:
Improve output clarity for attempting update/delete for empty habit list

wenxin-c committed:
Update wenxin PPP content

wenxin-c committed:
Fix typo in wenxin PPP

wenxin-c committed:
Merge pull request #294 from YongbinWang/update-developerguide

Update developerguide

wenxin-c committed:
Simplify self reflection sequence diagram

wenxin-c committed:
Merge branch 'master' into document-WenxinPPPUpdate

YongbinWang committed:
Implement printing for decrement as well to differe from increment printing

nichyjt committed:
Update focusmanager description

nichyjt committed:
Update error message for clarity

nichyjt committed:
Add UI-specific logic and break logical checks

nichyjt committed:
Use OS-agnostic line separator

haoyangw committed:
Merge branch 'master' into task-improveDgFormatting

haoyangw committed:
Standardise section headers
Replace 'Implementation Rationale' with
  'Design Considerations' to standardise
  these section headers.

haoyangw committed:
Merge pull request #293 from haoyangw/task-improveDgFormatting

Improve DeveloperGuide Formatting

haoyangw committed:
Merge branch 'master' into task-gamifDG

YongbinWang committed:
Resolve adding integer as habit name bug

nichyjt committed:
Fix formatting

nichyjt committed:
Update text messages to be user-friendly

nichyjt committed:
Update user guide regressions

nichyjt committed:
Merge branch 'master' into bugfix-FocusTimer-BugHunt

haoyangw committed:
Update DeveloperGuide Manager UML

YongbinWang committed:
Address Java CI and update atomichabit test, remove parseInt attempt

YongbinWang committed:
Merge branch 'master' into WellNusPE-atomichabit-bugfix

YongbinWang committed:
Update AtomicHabit sequence diagram for visual clarity and resolve deactivating helpcommand

nichyjt committed:
Update test regressions

YongbinWang committed:
Generate updated png file for atomichabit sequence

nichyjt committed:
Merge remote-tracking branch 'origin/bugfix-FocusTimer-BugHunt' into bugfix-FocusTimer-BugHunt

nichyjt committed:
Update test regressions and messages

nichyjt committed:
Remove unused import

wenxin-c committed:
Merge pull request #338 from YongbinWang/fix-sequencediagram-colour

Fix sequencediagram colour

wenxin-c committed:
Add quotation mark to questions

wenxin-c committed:
Fix error message error

wenxin-c committed:
Fix help command and ft help indentation error

wenxin-c committed:
Fix feature help message format

haoyangw committed:
Add UML Diagrams for Gamification

haoyangw committed:
Add gamification section to DeveloperGuide
Provide the implementation rationale for
  the gamification feature along with
  some alternative implementations that
  were considered(but dropped).

haoyangw committed:
Move GamificationTokenizer to correct package
Tokenizers belong in wellnus.storage, not the
  feature package, so move GamificationTokenizer
  from wellnus.gamification.util to wellnus.storage

haoyangw committed:
Change constructor access modifier
Now that GamificationTokenizer is in a different package,
  GamificationData(int xp) constructor needs to be
  public to allow GamificationTokenizer to initialise
  GamificationData as before.

haoyangw committed:
Fix unnecessary Exception thrown
GamificationTokenizer throws an Exception when
  the gamification data file is empty, which
  actually isn't an issue and we don't have to
  inform the user.

nichyjt committed:
Fix diagram errors

nichyjt committed:
Update user guide command syntax explanation

wenxin-c committed:
Standardise error notes across WellNUS++

nichyjt committed:
Update diagram colors

nichyjt committed:
Update content and fix TOC

nichyjt committed:
Update PPP for brevity

nichyjt committed:
Merge branch 'master' into bugfix-FocusTimer-BugHunt

nichyjt committed:
Merge branch 'master' into WellNusPE-atomichabit-bugfix

wenxin-c committed:
Merge pull request #336 from nichyjt/bugfix-FocusTimer-BugHunt

Fix `ft` usability bugs

YongbinWang committed:
Remove validateCommand throwing numberformatexception

YongbinWang committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-4/tp into WellNusPE-atomichabit-bugfix

YongbinWang committed:
Merge branch 'WellNusPE-atomichabit-bugfix' of https://github.com/YongbinWang/tp into WellNusPE-atomichabit-bugfix

YongbinWang committed:
Merge pull request #337 from YongbinWang/WellNusPE-atomichabit-bugfix

Well nus pe atomichabit bugfix

wenxin-c committed:
Resolve merge conflicts

wenxin-c committed:
Fix timer error message regression error

wenxin-c committed:
Merge pull request #339 from wenxin-c/bug-ReflectBugFix

Reflect error message and help command format bug fix

nichyjt committed:
Add strip logic to fix payload issue

haoyangw committed:
Update gamification error message
Shorten the error message section for
  Exceptions while loading gamification
  data from storage.

haoyangw committed:
Reorder MainManager execution order
Move setSupportedFeatureManagers() call into
  MainManager.executeCommands().
This ensures that the WellNUS++ logo and
  greeting is always printed first(especially
  if any exceptions occur).

haoyangw committed:
Merge branch 'master' into bugfix-gamifTokenizer

haoyangw committed:
Add missing EOF newline

haoyangw committed:
Merge pull request #340 from haoyangw/task-gamifDG

Add gamification section to DeveloperGuide

wenxin-c committed:
Merge pull request #342 from nichyjt/fix-CommandParserTrim

Add strip logic to fix payload issue

haoyangw committed:
Remove unused import

haoyangw committed:
Merge pull request #335 from wenxin-c/document-WenxinPPPUpdate

Update wenxin-c PPP

nichyjt committed:
Resolve merge conflicts

nichyjt committed:
Merge branch 'master' into docs-UpdateDevGuide-PPP-Nicholas

nichyjt committed:
Remove design considerations from focus timer

nichyjt committed:
Update TOC

nichyjt committed:
Merge pull request #341 from nichyjt/docs-UpdateDevGuide-PPP-Nicholas

Update `PPP` for Nicholas and various `DG`, `UG` fixes

haoyangw committed:
Improve WellNusLogger file logic
WellNusLogger allows opening multiple log
  files with the 'same' file name, which currently
  leads to the creation of logFile.log.1, logFile.log.2,
  etc. All these files are also not closed when WellNUS++
  is closed, which leaves logFile.log.lck files in
  storage.
Fix these issues by implementing singleton paradigm
  for the log file's associated FileHandler so that
  only one FileHandler is ever created and shared
  by the entire app. Also, close the FileHandler
  before WellNUS++ quits to remove all the
  logFile.log.lck files from storage.

haoyangw committed:
Close log file after quitting app
In MainManager, close the log file's
  FileHandler using WellNusLogger.closeLogFile()
  just before exiting WellNUS++ to
  remove the logFile.log.lck files from
  storage.

wenxin-c committed:
Summarize wenxin PPP

nichyjt committed:
Update PPP for brevity

wenxin-c committed:
Add pull request link

wenxin-c committed:
Fix ppp typo

haoyangw committed:
Merge pull request #346 from wenxin-c/document-ShortenPPP

Shorten wenxin's PPP more

nichyjt committed:
Update community contributions section

YongbinWang committed:
Update useguide to mention count initialised to 0 when added. Add Testing. Update private/public symbols.

nichyjt committed:
Add justification link for Enhancement

YongbinWang committed:
Merge branch 'master' into Fix-atomic-documentation

nichyjt committed:
Add link reference to enhancement reason

haoyangw committed:
Explain beep sound in UserGuide
Inform the user that a beep is played when
  the timer in focus timer is complete so
  he/she won't be caught surprised the first
  time he/she uses our app(and hears a beep).

haoyangw committed:
Explain Pomodoro Technique in UserGuide
Give a summary of the pomodoro technique
  in the UserGuide, but also include the
  link to a detailed explanation if the
  user wants to learn more about the
  technique.

haoyangw committed:
Improve gamification section of UserGuide
Give an example of how a user can gain
  XP points, which is using the update
  command of atomic habits.

haoyangw committed:
Improve Quick Start of UserGuide
Provide some context on where the users are
  when the first start WellNUS++.
Also, rephrase the Overview section to clearly
  but briefly explain different sessions within
  out app, aided by the beautiful diagram from
  @wenxin-c(thanks!).

haoyangw committed:
Improve topmost help of UserGuide
Rephrase the topmost help section to make it
  clearer.

BernardLesley committed:
Add checking for habit name and preserve habit order after tokenization

nichyjt committed:
Update documentation and diagrams

nichyjt committed:
Merge pull request #345 from nichyjt/docs-Nicholas-PPP-ReducePages

Update Nicholas' PPP for brevity

nichyjt committed:
Merge branch 'master' into Fix-atomic-documentation

nichyjt committed:
Merge branch 'master' into enhancement-cleanerLogging

BernardLesley committed:
Standardize tokenizer error

YongbinWang committed:
Merge pull request #347 from YongbinWang/Fix-atomic-documentation

Address documentation issue. Added testing.

BernardLesley committed:
Merge branch 'master' into BugFix-HabitNameChecking&StandardizeError

haoyangw committed:
Expose class attribute for JUnit Test
Expose GamificationData.POINTS_PER_LEVEL private
  attribute so that a JUnit Test can use it to
  write test cases flexibly(without hard coding).

haoyangw committed:
Add GamificationDataTest.class
Provide basic test cases for GamificationData's
  statistics helper methods to ensure that
  they are computing correctly.

nichyjt committed:
Truncate PPP further

nichyjt committed:
Fix grammar mistake

nichyjt committed:
Merge branch 'master' into docs-ShortenPPP-Nicholas

BernardLesley committed:
refactor ReflectionTokenizer and QuestionList for error handling

BernardLesley committed:
remove unused import

nichyjt committed:
Merge branch 'master' into bugfix-gamifTokenizer

haoyangw committed:
Fix wrong import order

haoyangw committed:
Define magic literals as constants

haoyangw committed:
Merge branch 'enhancement-cleanerLogging' of https://github.com/haoyangw/tp into enhancement-cleanerLogging

nichyjt committed:
Merge pull request #344 from haoyangw/enhancement-cleanerLogging

Cleaner logging

BernardLesley committed:
minor fix

haoyangw committed:
Redefine magic literals as class constants

haoyangw committed:
Merge branch 'bugfix-gamifTokenizer' of https://github.com/haoyangw/tp into bugfix-gamifTokenizer

wenxin-c committed:
Merge pull request #351 from nichyjt/docs-ShortenPPP-Nicholas

Shorten PPP to fit page limit

nichyjt committed:
Merge pull request #343 from haoyangw/bugfix-gamifTokenizer

Fix bogus Exception in GamificationTokenizer

wenxin-c committed:
Remove magic literal for UI test

BernardLesley committed:
erase unused variable

wenxin-c committed:
Merge pull request #352 from wenxin-c/task-RemoveUiTestMagicLiteral

Remove magic literal for UI test

BernardLesley committed:
remove unused import

haoyangw committed:
Merge branch 'master' into enhancement-userGuideCleanup

haoyangw committed:
Merge pull request #348 from haoyangw/enhancement-userGuideCleanup

User guide improvements/fixes

wenxin-c committed:
Merge pull request #349 from BernardLesley/BugFix-HabitNameChecking&StandardizeError

Bug fix habit name checking and standardize error

nichyjt committed:
Factor out magic literals

nichyjt committed:
Factor out magic literals and add JavaDoc

haoyangw committed:
Add data file cleaning method to gamif
Provide GamificationStorage.cleanDataFile() to allow
  cleaning the data file in cases such as data file
  corruption.

haoyangw committed:
Clean data file as needed
If gamification data file is found to be corrupted,
  use the new GamificationStorage.cleanDataFile()
  method to clean it and prevent future exceptions.

haoyangw committed:
Update gamification Class Diagram

haoyangw committed:
Move magic literal to class constant

haoyangw committed:
Merge branch 'master' into task-gamifTesting

nichyjt committed:
Factor out magic literals and add JavaDoc

nichyjt committed:
Add JavaDoc and authorship

nichyjt committed:
Merge branch 'master' into fix-CodingViolations-nichyjt

wenxin-c committed:
Add test javadoc for ui and reflect

wenxin-c committed:
Merge pull request #353 from haoyangw/bugfix-gamifTokenizerCorruptData

`GamificationTokenizer` corrupt data file cleaning

nichyjt committed:
Fix tests

nichyjt committed:
Merge remote-tracking branch 'origin/fix-CodingViolations-nichyjt' into fix-CodingViolations-nichyjt

wenxin-c committed:
Add sequence diagram cross

BernardLesley committed:
Handle overlooked edge case

BernardLesley committed:
revert unnecessary changes

wenxin-c committed:
Merge pull request #354 from nichyjt/fix-CodingViolations-nichyjt

Fix coding violations nichyjt

wenxin-c committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-4/tp

BernardLesley committed:
Fix CI failing

BernardLesley committed:
Merge branch 'master' into BufFix-ReflectTokenizerEdgeCase

wenxin-c committed:
Merge pull request #356 from BernardLesley/BufFix-ReflectTokenizerEdgeCase

Handle overlooked edge case in Reflect Tokenizer

wenxin-c committed:
Merge pull request #355 from wenxin-c/task-AddTestJavadoc

Add test javadoc for ui and reflect

wenxin-c committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T12-4/tp

wenxin-c committed:
Remove unused variables

wenxin-c committed:
Remove used method and duplicate catch

wenxin-c committed:
Update reflect sequence diagram

nichyjt committed:
Update TOC

nichyjt committed:
Update TOC and DG details

nichyjt committed:
Update diagrams

nichyjt committed:
Update authorship tags

nichyjt committed:
Update authorship tags

nichyjt committed:
Update authorship tags

wenxin-c committed:
Adjust page break for PPP

wenxin-c committed:
Merge pull request #358 from wenxin-c/bug-UpdateSequenceDiagram

Bug update sequence diagram

wenxin-c committed:
Summarise wenxin's PPP

haoyangw committed:
Add negative JUnit Tests
Test invalid inputs for GamificationData.addXp()
  and GamificationData.minusXp().

nichyjt committed:
Merge pull request #360 from wenxin-c/document-PPPFinalCut

Summarise wenxin's PPP

nichyjt committed:
Merge branch 'master' into task-ReorderDG

wenxin-c committed:
Merge pull request #359 from nichyjt/task-ReorderDG

Improve DG diagrams and partial ordering of DG

nichyjt committed:
Fix diagram errors

haoyangw committed:
Write JUnit tests for GamificationTokenizer

haoyangw committed:
Add input validation
Caught potential bug when GamificationTokenizer is
  detokenizing data from storage while running
  GamificationTokenizerTest.
Fix by adding input validation when parsing data
  read from storage.

haoyangw committed:
Add missing newline

haoyangw committed:
Add JavaDoc for GamificationDataTest

haoyangw committed:
Improve JUnit Test error messages
When unexpected exceptions are thrown, provide
  descriptive error messages to help with fixing
  the test cases.

haoyangw committed:
Merge branch 'master' into task-gamifTesting

BernardLesley committed:
Refactor tokenizer test

nichyjt committed:
Fix wording to force page break

haoyangw committed:
Add JavaDoc for GamificationTokenizerTest

haoyangw committed:
Add authorship tag for RepoSense

haoyangw committed:
Expose private methods for JUnit Tests
Expose MainManager.getMainCommandFor() and
  MainManager.getManagerFor() to allow writing JUnit Tests
  to verify their logic.

haoyangw committed:
Add MainManagerTest
Verify that important behaviours of MainManager
  function as intended and expected.

BernardLesley committed:
Improve tokenizer diagram

haoyangw committed:
Remove TODO comments

haoyangw committed:
Remove bad list index from UG

haoyangw committed:
Merge pull request #350 from haoyangw/task-gamifTesting

Gamification JUnit Testing

BernardLesley committed:
add new line

haoyangw committed:
Move magic literals to class constants

nichyjt committed:
Merge branch 'master' into Enhancement-TokenizerDiagram

nichyjt committed:
Merge pull request #364 from BernardLesley/Enhancement-TokenizerDiagram

Improve tokenizer diagram

nichyjt committed:
Merge pull request #365 from haoyangw/bugfix-codeCleanup

Cleanup code and UserGuide

wenxin-c committed:
Adjust PPP page break

nichyjt committed:
Update tagged issues

nichyjt committed:
Merge branch 'master' into task-ReadabilityPPP-Nicholas

nichyjt committed:
Merge pull request #366 from wenxin-c/bug-WenxinPPPFinalFix

Adjust PPP page break

nichyjt committed:
Merge pull request #362 from nichyjt/task-ReadabilityPPP-Nicholas

Add page break for PPP for Nicholas

haoyangw committed:
Update contributions in haoyangw's PPP

haoyangw committed:
Merge pull request #361 from BernardLesley/Enhancement-RefactorTokenizerTest

Refactor Tokenizer tests

nichyjt committed:
Fix output regression inaccuracies

YongbinWang committed:
Update PPP to summarize contributions

YongbinWang committed:
Merge branch 'master' into Fix-Shorten-PPP

nichyjt committed:
Fix obscure data directory deletion bug

haoyangw committed:
Merge pull request #369 from YongbinWang/Fix-Shorten-PPP

Shorten Yongbin's PPP

haoyangw committed:
Merge pull request #370 from nichyjt/fix-DataDirectoryDeleted

Fix obscure data directory deletion bug

wenxin-c committed:
Merge pull request #367 from haoyangw/task-improveHaoyangwPpp

Update contributions in haoyangw's PPP

nichyjt committed:
Fix minor regression issue

wenxin-c committed:
Merge pull request #363 from haoyangw/task-mainManagerTest

Write JUnit tests for `MainManager`

BernardLesley committed:
Add more explanation to PPP

wenxin-c committed:
Merge pull request #371 from BernardLesley/Enhancement-BernardPPP

Bernard PPP

nichyjt committed:
Fix filename

wenxin-c committed:
Merge pull request #368 from nichyjt/fix-UserGuide-Regressions

Fix output regression inaccuracies

haoyangw committed:
Fix typo

YongbinWang committed:
Update PPP for formatting error

haoyangw committed:
Link haoyangw/ped repo in haoyangw's PPP

YongbinWang committed:
Merge pull request #372 from YongbinWang/Fix-PPP-formatting

Update PPP for formatting error

nichyjt committed:
Fix typo

wenxin-c committed:
Resolve merge conflicts

wenxin-c committed:
Merge pull request #373 from haoyangw/bugfix-haoyangwPPPTypo

Fix typo in haoyangw's PPP

haoyangw committed:
Merge pull request #357 from wenxin-c/task-RemoveUnuseVariables

Remove unused variables

nichyjt committed:
Fix typo

nichyjt committed:
Fix code typo

haoyangw committed:
Fix typo in DG

haoyangw committed:
Update DG release jar name and URL

nichyjt committed:
Fix typo

haoyangw committed:
Update DG test cases' outputs

YongbinWang committed:
Remove extra space printed for MainManger when invalid command is given

YongbinWang committed:
Merge branch 'master' into Fix-invalidCommand-PrintingError

wenxin-c committed:
Merge pull request #375 from haoyangw/task-dgTestCaseUpdates

Update DeveloperGuide test cases' outputs

YongbinWang committed:
Update EXPECTED.TXT

wenxin-c committed:
Merge pull request #376 from nichyjt/fix-ConfigCommandMessage

Fix Config Command message typo

nichyjt committed:
Merge branch 'master' into fix-DeveloperGuide-Typos-CommandParser

wenxin-c committed:
Merge pull request #377 from YongbinWang/Fix-invalidCommand-PrintingError

Remove extra space printed for MainManger when invalid command is given

wenxin-c committed:
Merge pull request #374 from nichyjt/fix-DeveloperGuide-Typos-CommandParser

Fix developer guide typos command parser

nichyjt committed:
Fix broken link

haoyangw committed:
Fix single-letter variable name

nichyjt committed:
Merge pull request #378 from nichyjt/fix-PPP-BrokenLinks-Nicholas

Fix broken link

wenxin-c committed:
Merge pull request #379 from haoyangw/bugfix-codeStyle

Fix single-letter variable name

wenxin-c committed:
Add DG user stories

wenxin-c committed:
Merge branch 'master' into document-AddUserStoryDG

wenxin-c committed:
Fix typo

wenxin-c committed:
Merge pull request #380 from wenxin-c/document-AddUserStoryDG

Document add user story dg

haoyangw committed:
Provide more test cases in DG

haoyangw committed:
Merge branch 'master' into task-dgTestCases

haoyangw committed:
Merge pull request #381 from haoyangw/task-dgTestCases

More DG test cases

haoyangw committed:
Standardise capitalisation in DG

haoyangw committed:
Update DG Table of Contents

haoyangw committed:
Update Acknowledgements in DG

haoyangw committed:
Merge pull request #382 from haoyangw/task-fixDgToC

Update DeveloperGuide Table of Contents

haoyangw committed:
Update gradle output jar version

haoyangw committed:
Make instructions clearer

nichyjt committed:
Fix typo

haoyangw committed:
Fix list indices in DG

wenxin-c committed:
Merge pull request #383 from nichyjt/fix-UG-Quickstart

Fix User Guide Quickstart Typo

wenxin-c committed:
Merge pull request #385 from haoyangw/bugfix-dgListIndex

Fix list indices in DeveloperGuide

wenxin-c committed:
Merge pull request #384 from haoyangw/task-gradlev2.1Update

Update gradle output jar version

nichyjt committed:
Fix broken link

wenxin-c committed:
Merge pull request #386 from nichyjt/fix-BrokenUG-Link

Fix broken link

haoyangw committed:
Fix DG list indices

wenxin-c committed:
Merge pull request #387 from haoyangw/bugfix-dgIndices

Fix DG list indices

nichyjt committed:
Fix broken TOC

wenxin-c committed:
Merge pull request #388 from nichyjt/fix-WrongTOC-Name

Fix broken TOC Name
